src/FeedsCleanup/Program.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                Error("Path to the latest packages folder is not given");
+                return 1;
+            }
+
+            if (!Directory.Exists(ciLatestPackagesFolder))
+            {
+                Error("Folder {0} doesn't exist.", ciLatestPackagesFolder);
+                return 1;
+            }
+
+            var coherentPackageNames = Directory.GetFiles(coherentPackagesFolder, "*.nupkg").Select(p => Path.GetFileNameWithoutExtension(p));
+
+            if (!RemoveFromCI(coherentPackageNames, ciLatestPackagesFolder, whatIf))
+            {
+                return 2;
+            }
+
+            if (!RemoveFromdMyGet(coherentPackageNames, feed, apiKey, whatIf))
+            {
+                return 3;
+            }
+
+            return 0;
+        }
+
+        private static bool RemoveFromCI(IEnumerable<string> coherentPackageNames, string ciLatestPackagesFolder, bool whatif)
+        {
+            var latestPackageFolders = Directory.GetDirectories(ciLatestPackagesFolder);
+
+            var deleteCandiates = new List<string>();
+
+            foreach (var latestPackageFolder in latestPackageFolders)
+            {
+                var projectName = Path.GetFileName(latestPackageFolder);
+
+                var matchedVersion = new HashSet<string>(
+                    coherentPackageNames
+                        .Where(packageName => packageName.IndexOf(projectName) == 0)
+                        .Select(packageName => packageName.Substring(projectName.Length + 1)));
+
+                if (matchedVersion.Any())
+                {
+                    deleteCandiates.AddRange(Directory.GetDirectories(latestPackageFolder)
+                                             .Where(p => !matchedVersion.Contains(Path.GetFileName(p))));
+                }
+                else
+                {
+                    deleteCandiates.Add(latestPackageFolder);
+                    Warn("Project {0} doesn't exist in this Coherence build hence it will be deleted from {1}", projectName, ciLatestPackagesFolder);
+                }
+            }
+
+            foreach (var candiate in deleteCandiates)
+            {
+                Info("Delete folder {0}", candiate);
+
+                if (!whatif)
+                {
+                    Info("Actual delete!");
+                    Directory.Delete(candiate, recursive: true);
+                }
+            }
+
+            return true;
+        }
+
+        private static bool RemoveFromdMyGet(IEnumerable<string> coherentPackageNames,
+                                             string feed,
+                                             string apiKey,
+                                             bool whatif)
+        {
+            Console.WriteLine("Listing packages on {0}", feed);
+            var task = Task.Run(() =>
+            {
+                var repo = PackageRepositoryFactory.Default.CreateRepository(feed);
+                return repo.GetPackages().ToList();
```

Comment:
BTW this is presumably used only with the volatile feed, so there aren't really all that many packages on there to begin with...